CMakeLists.txt

```diff
@@ -114,13 +133,30 @@ if(ENABLE_LTO)
     set(CMAKE_INTERPROCEDURAL_OPTIMIZATION TRUE)
 endif()
 
-if(ENABLE_OPENMP)
-    find_package( OpenMP REQUIRED )
-    include(FindOpenMP)
-    if(OPENMP_FOUND)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
-        message(STATUS "found OpenMP, compiling with flags: " ${OpenMP_CXX_FLAGS} )
+# we have to look for openmp and set the include directory to prevent compilation errors.
+if (APPLE)
+    # set(OpenMP_CXX_LIB_NAMES "/usr/local/opt/libomp/lib/libomp.dylib")
+    # set(OpenMP_FOUND TRUE)
+    # set(OpenMP_CXX_FLAGS "-Xclang -fopenmp")
+    # include_directories("/usr/local/opt/libomp/include")
+    if(CMAKE_OSX_ARCHITECTURES EQUAL "arm64")
+        set(CMAKE_SYSTEM_PROCESSOR "aarch64")
+        set(LIBOMP_ARCH "aarch64")
+    else()
+        set(LIBOMP_ARCH "x86_64")
     endif()
+    set(OPENMP_STANDALONE_BUILD ON)
+    set(LIBOMP_ENABLE_SHARED OFF)
+    set(LIBOMP_INSTALL_ALIASES OFF)
+    add_subdirectory(extlib/llvm/openmp)
+else()
+    find_package(OpenMP REQUIRED)
```

Comment:
Yes, on macOS, this PR changes the behaviour to compile OpenMP by ourselves, so it can be statically linked, and so we can set the target architecture.